File: backend/app/schemas/source.py
from datetime import datetime
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel, HttpUrl


class SourceType(str, Enum):
    arxiv = "arxiv"
    github = "github"
    web = "web"


class SourceItemType(str, Enum):
    paper = "paper"
    repo = "repo"
    article = "article"
    product_update = "product_update"
    company_update = "company_update"


class SourceItem(BaseModel):
    id: str
    source: SourceType
    item_type: SourceItemType
    title: str
    url: Optional[HttpUrl] = None
    summary: str = ""
    authors: List[str] = []
    published_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    tags: List[str] = []
    extra: dict = {}


class SourceSearchRequest(BaseModel):
    query: str
    max_results: int = 5


class SourceSearchResponse(BaseModel):
    query: str
    source: SourceType
    items: List[SourceItem]
    fetched_at: datetime


class CombinedSearchResponse(BaseModel):
    query: str
    items: List[SourceItem]
    fetched_at: datetime

File: ios/InfraAgent/Models/Source.swift
import Foundation

struct SourceItem: Codable, Identifiable {
    let id: String
    let source: String
    let itemType: String
    let title: String
    let url: URL?
    let summary: String
    let authors: [String]
    let publishedAt: Date?
    let updatedAt: Date?
    let tags: [String]

    enum CodingKeys: String, CodingKey {
        case id
        case source
        case itemType = "item_type"
        case title
        case url
        case summary
        case authors
        case publishedAt = "published_at"
        case updatedAt = "updated_at"
        case tags
    }
}

struct SourceSearchResponse: Codable {
    let query: String
    let source: String
    let items: [SourceItem]
    let fetchedAt: Date

    enum CodingKeys: String, CodingKey {
        case query
        case source
        case items
        case fetchedAt = "fetched_at"
    }
}

struct CombinedSearchResponse: Codable {
    let query: String
    let items: [SourceItem]
    let fetchedAt: Date

    enum CodingKeys: String, CodingKey {
        case query
        case items
        case fetchedAt = "fetched_at"
    }
}
